pilot/pkg/trustbundle/trustbundle.go

```diff
@@ -0,0 +1,181 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package trustbundle
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"strings"
+	"sync"
+
+	meshconfig "istio.io/api/mesh/v1alpha1"
+	"istio.io/pkg/log"
+)
+
+// source is all possible sources of MeshConfig
+type Source int
+
+type TrustAnchorConfig struct {
+	Source Source
+	Certs  []string
+}
+
+type TrustAnchorUpdate struct {
+	TrustAnchorConfig
+}
+
+type TrustBundle struct {
+	sourceConfig map[Source]TrustAnchorConfig
+	mutex        sync.RWMutex
+	mergedCerts  []string
+	updatecb     func()
+}
+
+var (
+	trustBundleLog = log.RegisterScope("trustBundle", "Workload mTLS trust bundle logs", 0)
+
+	// GlobalWorkloadTrustbundle Needed to interface with XDS
+	GlobalWorkloadTrustbundle *TrustBundle = nil
```

Comment:
how is this going to be used? Generally global variable like this is pretty suspicious. Ill check the full PR

Reply:
Looking at the full code, you should be able to just pass the TrustBundle to PcdsGenerator and avoid making it global

Reply:
Have added comments and locks when setting the globalWorkloadTrustBundle, indicating that this should only be done once. Will that suffice? I can pass this to the Pcds Generator, but it won't be as neat - will require passing this argument through a stack of functions until initGenerator()

Reply:
I think some dependency passing is probably preferably to having global variables

Reply:
what about an internal package variable that needs to be explicitly set? This is similar to the way the spiffee package sets and gets the TrustDomain(). That would be an indicator that it isn't meant to be accessed from outside the package without good cause?